client/src/modules/Tasks/pages/TasksPage/TasksPage.tsx

```diff
@@ -0,0 +1,146 @@
+  };
+
+  const handleModalSubmit = useCallback(
+    async (values: any) => {
+      const checkCriteria = () => {
+        return dataCriteria.every(item => {
+          if (item.type !== TaskType.Title) {
+            return item.max !== 0;
+          }
+          return true;
+        });
+      };
+
+      const isVerified = dataCriteria.length ? checkCriteria() : true;
+      if (!isVerified) {
+        message.error('Please, check criteria! It has subtask with no score.');
+        return;
+      }
+
+      try {
+        if (modalLoading) {
+          return;
+        }
+        setModalLoading(true);
+        const record = createRecord(values);
+        if (modalAction === 'update') {
+          await tasksApi.updateTask(modalData!.id!, record);
+          const { data } = await criteriaApi.getTaskCriteria(modalData!.id!);
+          if (data.criteria) {
+            await criteriaApi.updateTaskCriteria(modalData!.id!, { criteria: dataCriteria });
+          } else {
+            await criteriaApi.createTaskCriteria(modalData!.id!, { criteria: dataCriteria });
+          }
+        } else {
+          const { data: task } = await tasksApi.createTask(record);
+          await criteriaApi.createTaskCriteria(task.id, { criteria: dataCriteria });
+        }
+        setModalData(null);
+      } catch (e) {
+        message.error('An error occurred. Please try again later.');
+      } finally {
+        setModalLoading(false);
+      }
+    },
+    [modalData, modalAction, modalLoading, dataCriteria],
+  );
+
+  return (
+    <AdminPageLayout title="Manage Tasks" loading={loading} courses={courses}>
+      <Content style={{ margin: 8 }}>
+        <Button type="primary" onClick={handleAddItem}>
+          Add Task
+        </Button>
+        <TableView data={data} handleEditItem={handleEditItem} />
+      </Content>
+      {modalData && (
+        <TaskModal
+          tasks={data}
+          dataCriteria={dataCriteria}
+          disciplines={disciplines}
+          handleModalSubmit={handleModalSubmit}
+          modalData={modalData}
+          modalLoading={modalLoading}
+          setDataCriteria={setDataCriteria}
+          setModalData={setModalData}
+          setModalValues={setModalValues}
+        />
+      )}
+    </AdminPageLayout>
+  );
+}
+
+function createRecord(values: any) {
+  const data: CreateTaskDto = {
+    type: values.type,
+    name: values.name,
+    githubPrRequired: !!values.githubPrRequired,
+    descriptionUrl: values.descriptionUrl,
+    githubRepoName: values.githubRepoName,
+    sourceGithubRepoUrl: values.sourceGithubRepoUrl,
+    description: values.description,
+    tags: values.tags,
+    skills: values.skills?.map((skill: string) => skill.toLowerCase()),
+    disciplineId: values.discipline,
+    attributes: JSON.parse(values.attributes ?? '{}') as Record<string, unknown>,
+  };
+  return data;
+}
+
+function prepareValues(modalData: TaskDto) {
+  if (!modalData) {
+    return modalData;
+  }
```

Comment:
modal data is declared with the type taskDto, as I understand it, this if block is not needed

Reply:
I find it strange too, and for 99% sure it is not needed, but that 1%)) 
Considering it's just a refactoring I left it as it is, just in case...

Reply:
Isn't that the point of refactoring, to make the code more understandable?))
If there can be null or undefined here, let's indicate it in the argument typing.

Reply:
I took another look at it, and it could be only a `TaskDto` )) 

removed it